optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/scope/DefaultSolverScope.java

```diff
@@ -47,15 +48,15 @@
      */
     protected Semaphore runnableThreadSemaphore = null;
 
-    protected volatile Long startingSystemTimeMillis;
-    protected volatile Long endingSystemTimeMillis;
+    private volatile Long startingSystemTimeNanos;
+    private volatile Long endingSystemTimeNanos;
     protected long childThreadsScoreCalculationCount = 0;
 
     protected Score startingInitializedScore;
 
     protected volatile Solution_ bestSolution;
     protected volatile Score bestScore;
-    protected Long bestSolutionTimeMillis;
+    private Long bestSolutionTimeNanos;
```

Comment:
I converted the class to use nanos instead of millis. No need not to use the precision, if we have it.
Also, I've hidden the fields, since they have getters and setters and this is not performance-sensitive code.